fuel-vm/src/checked_transaction.rs

```diff
@@ -288,7 +288,13 @@ pub enum CheckError {
     /// The transaction doesn't pass validity rules.
     Validity(ValidityError),
     /// The predicate verification failed.
-    PredicateVerificationFailed(PredicateVerificationFailed),
+    PredicateVerificationFailed {
+        /// Input index of the failed predicate.
+        /// `TransactionExceedsTotalGasAllowance` sets this to `0`.
+        input_index: usize,
```

Comment:
If predicate index can only be u16 maybe we could use u16::MAX + 1 for general error. Or have an option. Because, currently 0 can be general or predicate 0 which is two different things

Reply:
That actually just felt too ugly. Instead I just rewrote the whole thing in 223e24819. to include the fields directly into  `PredicateVerificationFailed` variants.